spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientSecretsCache.java

```diff
@@ -81,7 +82,8 @@ private static List<StrippedSourceContainer> strippedSecrets(List<V1Secret> secr
 	}
 
 	private static Map<String, String> transform(Map<String, byte[]> in) {
-		return in.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, en -> new String(en.getValue())));
+		return ObjectUtils.isEmpty(in) ? Map.of() :
```

Comment:
did not expect this one :| I guess reproduced somewhere for you internally on a project?

Reply:
yup ran into it when i was trying to looking into the masking secrets PR